Better CDS Lint documentation (#756)
cap-js/docs: tools/lint/meta/auth-restrict-grant-service.md

---
editLink: false
outline: [2,2]
breadcrumbs:
  - CDS Lint
    - Rules Reference
status: released
---

<script setup>
  import PlaygroundBadge from '../../../.vitepress/theme/components/PlaygroundBadge.vue'
</script>

# auth-restrict-grant-service

## Rule Details

> Restrictions can be defined on different types of CDS resources, but there are some limitations with regards to supported privileges (see [limitations](../../../guides/security/authorization#supported-combinations-with-cds-resources)). Unsupported privilege properties are ignored by the runtime.

Reviewer: This sentence starts the next paragraph.
```suggestion
Restrictions can be defined on different types of CDS resources, but there are some limitations with regards to supported privileges (see [limitations](../../../guides/security/authorization#supported-combinations-with-cds-resources)).
```

> Unsupported privilege properties are ignored by the runtime. Especially, for bound or unbound actions, the `grant` property is implicitly removed (assuming grant: '*' instead). The same also holds for functions. This rule ensures that `@restrict.grant` on service level and for bound/unbound actions and functions is limited to `grant: '*'`.

Reviewer: ```suggestion
Unsupported privilege properties are ignored by the runtime. Especially, for bound or unbound actions, the `grant` property is implicitly removed (assuming `grant: '*'` instead). The same is true for functions. This rule ensures that `@restrict.grant` on service level and for bound/unbound actions and functions is limited to `grant: '*'`.
```

---
editLink: false
outline: [2,2]
breadcrumbs:
  - CDS Lint
    - Rules Reference
status: released
---

<style>
.vp-code {
  overflow-x: hidden !important;
}
</style>

<script setup>
  import PlaygroundBadge from '../../../.vitepress/theme/components/PlaygroundBadge.vue'
</script>

# auth-restrict-grant-service

## Rule Details

Restrictions can be defined on different types of CDS resources, but there are some limitations with regards to supported privileges (see [limitations](../../../guides/security/authorization#supported-combinations-with-cds-resources)). Unsupported privilege properties are ignored by the runtime.

> An unsupported privilege is that for bound or unbound actions, the `grant` property is implicitly removed (assuming grant: '*' instead). The same also holds for functions. This rule ensures that `@restrict.grant` on service level and for bound/unbound actions and functions is limited to `grant: '*'`.

Reviewer 2: "is that" sound like German. Could you rephrase it?

Author: You are right, I just updated it to be phrased identically to the *Authorization* docs.

### Examples

#### ✅ &nbsp; Correct example

> Let's consider the following example with the `CatalogService` where there function `getViewsCount()` is restricted to the *Admin* role, granting all CDS events:

Reviewer 2: ```suggestion
Let's consider the following example with the `CatalogService` where the function `getViewsCount()` is restricted to the *Admin* role, granting all CDS events:
```
